Toolhouse.Monitoring/Metrics.cs

```diff
@@ -157,77 +155,6 @@ public static void ObserveHttpRequestDuration(TimeSpan duration)
                 ).Observe(duration.TotalSeconds);
         }
 
-        /// <summary>
-        /// Instruments a block of code that makes a request to an external API (e.g. via REST / SOAP).
-        /// </summary>
-        /// <param name="name">Descriptive name of thing being done. Should be a singular noun, e.g. "salesforce".</param>
-        /// <param name="callback">Code to run. Should return true for success, false for failure.</param>
-        public static void InstrumentApiCall(string name, Func<bool> makeRequest)
```

Comment:
We should leave this method in place, but rework it to use the new instrumentation code under the hood.